### ISSUE TITLE
Unknown Parse UI Login Failure

### ISSUE THREAD

**Issue by StuStirling** [ParsePlatform/ParseUI-Android#111]
I have a user who has logged in fine on one device using an email account but on another he can't log in at all and gets the toast "Could not login, please try again later.".

He assures me he is using the correct credentials.

I have tracked down where this toast gets launched from and its in the com.parse.ui.ParseLoginFragment.

The line in question is called when the login returns an error with an error code that does not match the  error code `ParseException.OBJECT_NOT_FOUND`.

    ParseUser.logInInBackground(username, password, new LogInCallback() {
            @Override
            public void done(ParseUser user, ParseException e) {
              if (isActivityDestroyed()) {
                return;
              }

              if (user != null) {
                loadingFinish();
                loginSuccess();
              } else {
                loadingFinish();
                if (e != null) {
                  debugLog(getString(R.string.com_parse_ui_login_warning_parse_login_failed) +
                      e.toString());
                  if (e.getCode() == ParseException.OBJECT_NOT_FOUND) {
                    if (config.getParseLoginInvalidCredentialsToastText() != null) {
                      showToast(config.getParseLoginInvalidCredentialsToastText());
                    } else {
                      showToast(R.string.com_parse_ui_parse_login_invalid_credentials_toast);
                    }
                    passwordField.selectAll();
                    passwordField.requestFocus();
                  } else {
                    showToast(R.string.com_parse_ui_parse_login_failed_unknown_toast);
                  }
                }
              }
            }
          });

**Comment by wangmengyan95** [comment]
Hi @StuStirling, for login, the ParseUI module just use a normal android `ParseUser.logInInBackground()` API call. Is it possible for you try the android or [rest](https://www.parse.com/docs/rest/guide#users-logging-in) API call to see what happens?

**Comment by StuStirling** [comment]
Not possible as this is in release and the user reporting it is still having the issue. I can tell the error code and message it's coming back with though as in a beta I released to him he has given me feedback.

    (100) i/o failure

**Comment by wangmengyan95** [comment]
Hi @StuStirling, is it possible for you to give us the full stack trace?  I/O failure probably means your user has the problem on connecting to Parse or facebook server, but without the stacktrace or a sample repo project, it is very hard for us to tell you the exact issue.

**Comment by chuonglam** [comment]
I'm having this issue too. I still don't know how to fix it.

**Comment by mliell** [comment]
I have the same problem, someone managed to solve?

**Comment by grantland** [comment]
@chuonglam @mliell could either of you update us on the stack trace?

**Comment by mliell** [comment]
in Logcat: 

ResourceType: Failure getting entry for 0x010802c9 (t=7 e=713) in package 0 (error -75)

**Comment by chuonglam** [comment]
- I simply use query: countInBackground, and here's the result, exception code and message:
count: -1
exception code: 100
message: i/o failure
312-370/system_process I/qtaguid: Failed write_ctrl(s 0 10021) res=-1 errno=1
- error 100 dedicates that device cannot connect to server, but I'm sure that my internet connection is fine.

**Comment by mliell** [comment]
When analyze android monitor network by clicking create account no data is fired or received.

**Comment by grantland** [comment]
@chuonglam @mliell We need the full stack trace, not just the message and code. An example here: https://github.com/ParsePlatform/Parse-SDK-Android/issues/326#issuecomment-168273988

The Parse SDK version, ParseUI version, and concise repro steps would be needed as well.

**Comment by mliell** [comment]
@grantland  #https://gist.github.com/mliell/925cc3725f1f02f4f066

**Comment by grantland** [comment]
@mliell there are no exception in those logs, so I'm not sure you're having the same issue described here. It seems you are having issues using Parse, which would be better asked on [Google Groups](https://groups.google.com/forum/#!forum/parse-developers) or [Stack Overflow](http://stackoverflow.com/tags/parse.com) as we try to reserve Github issues for bugs discovered in the SDK.

**Comment by chuonglam** [comment]
hi @grantland, I think my error is fixed by ParsePlatform/Parse-SDK-Android#326 (comment) now, thank you for your support.

**Comment by StuStirling** [comment]
Here is the stack trace for when there isn't a valid internet connection. I got excited and thought I was experiencing the same issue as my users but then I realised that the simulator's WiFi had gone wrong :/

```
01-25 16:20:06.838 4664-4664/com.corecoders.skitracks W/System.err: com.parse.ParseRequest$ParseRequestException: i/o failure
01-25 16:20:06.838 4664-4664/com.corecoders.skitracks W/System.err:     at com.parse.ParseRequest.newTemporaryException(ParseRequest.java:321)
01-25 16:20:06.838 4664-4664/com.corecoders.skitracks W/System.err:     at com.parse.ParseRequest$2.then(ParseRequest.java:159)
01-25 16:20:06.838 4664-4664/com.corecoders.skitracks W/System.err:     at com.parse.ParseRequest$2.then(ParseRequest.java:153)
01-25 16:20:06.838 4664-4664/com.corecoders.skitracks W/System.err:     at bolts.Task$15.run(Task.java:825)
01-25 16:20:06.838 4664-4664/com.corecoders.skitracks W/System.err:     at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1112)
01-25 16:20:06.838 4664-4664/com.corecoders.skitracks W/System.err:     at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:587)
01-25 16:20:06.838 4664-4664/com.corecoders.skitracks W/System.err:     at java.lang.Thread.run(Thread.java:818)
01-25 16:20:06.838 4664-4664/com.corecoders.skitracks W/System.err: Caused by: java.net.UnknownHostException: Unable to resolve host "api.parse.com": No address associated with hostname
01-25 16:20:06.841 4664-4664/com.corecoders.skitracks W/System.err:     at java.net.InetAddress.lookupHostByName(InetAddress.java:457)
01-25 16:20:06.841 4664-4664/com.corecoders.skitracks W/System.err:     at java.net.InetAddress.getAllByNameImpl(InetAddress.java:252)
01-25 16:20:06.841 4664-4664/com.corecoders.skitracks W/System.err:     at java.net.InetAddress.getAllByName(InetAddress.java:215)
01-25 16:20:06.841 4664-4664/com.corecoders.skitracks W/System.err:     at com.android.okhttp.HostResolver$1.getAllByName(HostResolver.java:29)
01-25 16:20:06.841 4664-4664/com.corecoders.skitracks W/System.err:     at com.android.okhttp.internal.http.RouteSelector.resetNextInetSocketAddress(RouteSelector.java:232)
01-25 16:20:06.841 4664-4664/com.corecoders.skitracks W/System.err:     at com.android.okhttp.internal.http.RouteSelector.next(RouteSelector.java:124)
01-25 16:20:06.841 4664-4664/com.corecoders.skitracks W/System.err:     at com.android.okhttp.internal.http.HttpEngine.connect(HttpEngine.java:272)
01-25 16:20:06.841 4664-4664/com.corecoders.skitracks W/System.err:     at com.android.okhttp.internal.http.HttpEngine.sendRequest(HttpEngine.java:211)
01-25 16:20:06.841 4664-4664/com.corecoders.skitracks W/System.err:     at com.android.okhttp.internal.http.HttpURLConnectionImpl.execute(HttpURLConnectionImpl.java:382)
01-25 16:20:06.841 4664-4664/com.corecoders.skitracks W/System.err:     at com.android.okhttp.internal.http.HttpURLConnectionImpl.connect(HttpURLConnectionImpl.java:106)
01-25 16:20:06.841 4664-4664/com.corecoders.skitracks W/System.err:     at com.android.okhttp.internal.http.HttpURLConnectionImpl.getOutputStream(HttpURLConnectionImpl.java:217)
01-25 16:20:06.841 4664-4664/com.corecoders.skitracks W/System.err:     at com.android.okhttp.internal.http.DelegatingHttpsURLConnection.getOutputStream(DelegatingHttpsURLConnection.java:218)
01-25 16:20:06.841 4664-4664/com.corecoders.skitracks W/System.err:     at com.android.okhttp.internal.http.HttpsURLConnectionImpl.getOutputStream(HttpsURLConnectionImpl.java:25)
01-25 16:20:06.841 4664-4664/com.corecoders.skitracks W/System.err:     at com.parse.ParseURLConnectionHttpClient.executeInternal(ParseURLConnectionHttpClient.java:51)
01-25 16:20:06.841 4664-4664/com.corecoders.skitracks W/System.err:     at com.parse.ParseHttpClient$ParseNetworkInterceptorChain.proceed(ParseHttpClient.java:158)
01-25 16:20:06.841 4664-4664/com.corecoders.skitracks W/System.err:     at com.parse.ParsePlugins$1.intercept(ParsePlugins.java:115)
01-25 16:20:06.841 4664-4664/com.corecoders.skitracks W/System.err:     at com.parse.ParseHttpClient$ParseNetworkInterceptorChain.proceed(ParseHttpClient.java:147)
01-25 16:20:06.841 4664-4664/com.corecoders.skitracks W/System.err:     at com.parse.ParseHttpClient.execute(ParseHttpClient.java:122)
01-25 16:20:06.841 4664-4664/com.corecoders.skitracks W/System.err:     at com.parse.ParseRequest$3.then(ParseRequest.java:150)
01-25 16:20:06.841 4664-4664/com.corecoders.skitracks W/System.err:     at com.parse.ParseRequest$3.then(ParseRequest.java:147)
01-25 16:20:06.841 4664-4664/com.corecoders.skitracks W/System.err:     at bolts.Task$15.run(Task.java:825)
01-25 16:20:06.841 4664-4664/com.corecoders.skitracks W/System.err:     at bolts.BoltsExecutors$ImmediateExecutor.execute(BoltsExecutors.java:105)
01-25 16:20:06.841 4664-4664/com.corecoders.skitracks W/System.err:     at bolts.Task.completeAfterTask(Task.java:816)
01-25 16:20:06.841 4664-4664/com.corecoders.skitracks W/System.err:     at bolts.Task.continueWithTask(Task.java:628)
01-25 16:20:06.841 4664-4664/com.corecoders.skitracks W/System.err:     at bolts.Task.continueWithTask(Task.java:639)
01-25 16:20:06.841 4664-4664/com.corecoders.skitracks W/System.err:     at bolts.Task$13.then(Task.java:731)
01-25 16:20:06.841 4664-4664/com.corecoders.skitracks W/System.err:     at bolts.Task$13.then(Task.java:719)
01-25 16:20:06.841 4664-4664/com.corecoders.skitracks W/System.err: 	... 4 more
01-25 16:20:06.841 4664-4664/com.corecoders.skitracks W/System.err: Caused by: android.system.GaiException: android_getaddrinfo failed: EAI_NODATA (No address associated with hostname)
01-25 16:20:06.841 4664-4664/com.corecoders.skitracks W/System.err:     at libcore.io.Posix.android_getaddrinfo(Native Method)
01-25 16:20:06.841 4664-4664/com.corecoders.skitracks W/System.err:     at libcore.io.ForwardingOs.android_getaddrinfo(ForwardingOs.java:55)
01-25 16:20:06.841 4664-4664/com.corecoders.skitracks W/System.err:     at java.net.InetAddress.lookupHostByName(InetAddress.java:438)
01-25 16:20:06.841 4664-4664/com.corecoders.skitracks W/System.err: 	... 30 more
```
